libs/gl-client/src/pairing/attestation_device.rs

```diff
@@ -0,0 +1,265 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use super::{into_approve_pairing_error, into_verify_pairing_data_error, Error};
+use crate::{
+    credentials::{RuneProvider, TlsConfigProvider},
+    pb::scheduler::{
+        pairing_client::PairingClient, ApprovePairingRequest, Empty, GetPairingDataRequest,
+        GetPairingDataResponse,
+    },
+    tls::TlsConfig,
+};
+use bytes::BufMut as _;
+use picky::{pem::Pem, x509::Csr};
+use picky_asn1_x509::{PublicKey, SubjectPublicKeyInfo};
+use ring::{
+    rand,
+    signature::{self, EcdsaKeyPair, KeyPair},
+};
+
+use rustls_pemfile as pemfile;
+use tonic::transport::Channel;
+
+type Result<T, E = super::Error> = core::result::Result<T, E>;
+
+pub struct Connected(PairingClient<Channel>);
+pub struct Unconnected();
+
+pub struct Client<T> {
+    inner: T,
+    tls: TlsConfig,
```

Comment:
Any strong objections to replacing `tls`, `rune`, and key with `creds`?

It would make the definition of the Client smaller and allows us to skip passing the node_id to `approve_pairing`